core/lib/dal/src/data_availability_dal.rs

```diff
@@ -0,0 +1,221 @@
+use zksync_db_connection::{
+    connection::Connection,
+    error::DalResult,
+    instrument::{InstrumentExt, Instrumented},
+};
+use zksync_types::{pubdata_da::DataAvailabilityBlob, L1BatchNumber};
+
+use crate::{
+    models::storage_data_availability::{L1BatchDA, StorageDABlob},
+    Core,
+};
+
+#[derive(Debug)]
+pub struct DataAvailabilityDal<'a, 'c> {
+    pub(crate) storage: &'a mut Connection<'c, Core>,
+}
+
+impl DataAvailabilityDal<'_, '_> {
+    /// Inserts the blob_id for the given L1 batch. If the blob_id is already present,
+    /// verifies that it matches the one provided in the function arguments
+    /// (preventing the same L1 batch from being stored twice)
+    pub async fn insert_l1_batch_da(
+        &mut self,
+        number: L1BatchNumber,
+        blob_id: &str,
+        sent_at: chrono::NaiveDateTime,
+    ) -> DalResult<()> {
+        let update_result = sqlx::query!(
+            r#"
+            INSERT INTO
+                data_availability (l1_batch_number, blob_id, sent_at, created_at, updated_at)
+            VALUES
+                ($1, $2, $3, NOW(), NOW())
+            "#,
+            i64::from(number.0),
+            blob_id,
+            sent_at,
+        )
+        .instrument("insert_l1_batch_da")
+        .with_arg("number", &number)
+        .with_arg("blob_id", &blob_id)
+        .report_latency()
+        .execute(self.storage)
+        .await?;
+
+        if update_result.rows_affected() == 0 {
```

Comment:
Stupid question, but don't you need `ON CONFLICT (l1_batch_number) DO NOTHING` for this logic to work?

Reply:
1. renamed and added a comment
2. yes, you are right, added it